history.py

```diff
@@ -0,0 +1,31 @@
+from django_mako_plus import view_function, jscontext
+from catalog import models as cmod
+
+class LastFiveMiddleware:
+    def __init__(self, get_response):
+        self.get_response = get_response
+        # One-time configuration and initialization.
+
+    def __call__(self, request):
+        # Code to be executed for each request before
+        # the view (and later middleware) are called.
+        product_ids = request.session.get('LastFiveList', [])
+        product_objs = []
+
+        for item in product_ids:
+            product_objs.append(cmod.Product.objects.get(id = item))
+
+        request.LastFiveList = product_objs
+
+        response = self.get_response(request)
+        # Code to be executed for each request/response after
+        # the view is called.
+
+        return_ids = []
+
+        for item2 in request.LastFiveList:
+            return_ids.append(item2.id)

```

Comment:
Silly Blake, I would suggest naming your variables something more descriptive than "item2" such as "kyleIsTheBestHeroEver_Array"

Reply:
I can't believe I missed this! I even made the same comment on a colleague's code.